lib/puppet/transaction/additional_resource_generator.rb

```diff
@@ -102,16 +103,69 @@ def add_resources(generated, resource)
     end
   end
 
-  def add_resource(res, parent_resource, priority)
+  def add_resource(res, parent_resource, priority=nil)
```

Comment:
Is the priority argument still used anywhere? If it's preserved only for backward compatibility, add a comment.